forms/api/payment.php

```
@@ -0,0 +1,158 @@
+
+            upsert_payment(
+                $payment_id,
+                $order['id'],
+                $charge['id'],
+                $payment_status,
+                (float) $order['amount'],
+                Currency::from($order['currency']),
+                PaymentType::from($input['payment_type']),
+                $charge['source']['type'],
+                $user_message,
+                json_encode($charge),
```

PHPStan (2.1.46)

[warning] Parameter `#10` $json_data of function upsert_payment expects string|null, string|false given.


(argument.type)